site/content/usage/06-vpc-networking.md

```diff
@@ -160,3 +160,32 @@ See the complete example [here](https://github.com/weaveworks/eksctl/blob/master
 
 **Note**: Specifying the NAT Gateway is only supported during cluster creation and it is not touched during a cluster
 upgrade. There are plans to support changing between different modes on cluster update in the future.
+
+### Managing Access to the Kubernetes API Server Endpoints
+
+The default creation of an EKS cluster exposes the Kubernetes API server publicly but not directly
+from within the
+VPC subnets (Public=true, Private=false). Traffic destined for the API server from within the VPC
+must first exit
+the VPC networks (but not Amazon's network) and then re-enter to reach the API server. Allowing direct access from
+within the VPC subnets is possible but not without some caveats. See [Amazon EKS Cluster
+Endpoint Access Control](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html) for
+further information
+on additional steps needed when public=false, private=true.
+
+The Kubernetes API server endpoint access for a cluster can be cofigured for public and private
+access by
+specifiying the `--vpc-endpoint-access-public` and/or`--vpc-endpoint-access-private` CLI flags to be
+`true` (access enabled)
+or
+`false` (access disabled).
+
+Alternatively, one can configure access in the cluster config file like
+the example below:
+
+```yaml
+vpc:
+  endpointAccess:
+    public:  <true|false>
```

Comment:
One alternative I can think of is to make it a string field, e.g. `vpc.clusterEndpoints: PrivateOnly`, `vpc.clusterEndpoints: Default`, `vpc.clusterEndpoints: PublicOnly`... but let's not rush this decision just yet.

Reply:
I think, in this scheme, we'd also need to have vpc.clusterEndpoints: All (or PublicPrivate).  And we'd defer PrivateOnly until we resolve how to deal with the times when we need kubernetes API access.

Reply:
Let's stick to boolean, but change the parent key to `clusterEndpoints`.

Reply:
Maybe just a slight tweak:
```
clusterEndpointsAccess:
    Public: true
    Private: false

Reply:
I'm going to start with the first option (clusterEndpointsAccess), but it should be an easy change to make if we want to go with something else.

Reply:
Actually, I'm now thinking of a slightly different naming scheme, to correspond with the EKS API names:
```
clusterEndpoints:
     PublicAccess: <bool>
     PrivateAccess: <bool>
```
Where the API names are:
1. EndpointPublicAccess
1. EndpointPrivateAccess

Again, not too difficult to rename if necessary.